### PR TITLE
Small fixes to the iOS build

### PR DESCRIPTION
**Pull request by badboy**
This will actually likely currently fail on CI, because the `swiftlint` we use there is older?
I have swiftlint 0.57 locally.

We might wanna update that check to Xcode 16 soon.

### CODE REVIEW

**Comment by badboy** [comment]
Oh it actually doesn't fail, because CI uses swiftlint 0.53 which is so old, it doesn't have that (somewhat broken) new lint!

**Review by travis79 — CHANGES_REQUESTED** [review]
Why are we reverting this when it was just changed to this 3 months ago? 

ref: https://github.com/mozilla/glean/commit/ad9855581e06c0190eabd1c2ebe06ca2eeefaa9a

**Comment by badboy** [comment]
> Why are we reverting this when it was just changed to this 3 months ago?
> 
> ref: [ad98555](https://github.com/mozilla/glean/commit/ad9855581e06c0190eabd1c2ebe06ca2eeefaa9a)

Because I should have included the link: https://github.com/realm/SwiftLint/issues/5263#issuecomment-2115182747

tl;dr: the way we changed it to is swallowing errors, and thus swiftlint actually reversed the decision and lints AGAINST this now.
That only happens with a newer swiftlint than we use on CI, but I have that newer one locally and thus `make lint-swift` fails for me. Given that we eventually upgrade CI and it doesn't fail right now we can land this.
